packages/ui/core-components/src/lib/unsorted/viz/table/TableRow.svelte

```diff
@@ -54,19 +54,48 @@
 		{#if $props.columns.length > 0}
 			{#each $props.columns.sort((a, b) => finalColumnOrder.indexOf(a.id) - finalColumnOrder.indexOf(b.id)) as column, k}
 				{@const useCol = safeExtractColumn(column, columnSummary)}
-				{@const column_min = column.colorMin ?? useCol.columnUnitSummary.min}
-				{@const column_max = column.colorMax ?? useCol.columnUnitSummary.max}
+				{@const scaleCol = column.scaleColumn
+					? columnSummary.find((d) => d.id === column.scaleColumn)
+					: useCol}
+				{@const column_min = column.colorMin ?? scaleCol.columnUnitSummary.min}
+				{@const column_max = column.colorMax ?? scaleCol.columnUnitSummary.max}
 				{@const is_nonzero =
 					column_max - column_min !== 0 && !isNaN(column_max) && !isNaN(column_min)}
 				{@const column_format = column.fmt
 					? getFormatObjectFromString(column.fmt, useCol.format?.valueType)
-					: useCol.format}
+					: column.fmtColumn
+						? getFormatObjectFromString(row[column.fmtColumn], useCol.format?.valuetype)
+						: useCol.format}
 				{@const color_domain =
 					column.colorBreakpoints ??
 					(column.colorMid ? [column_min, column.colorMid, column_max] : [column_min, column_max])}
 				{@const color_scale = column.colorPalette
-					? chroma.scale(column.colorPalette).domain(color_domain)
+					? chroma.scale(column.colorPalette).domain(color_domain).nodata('white')
 					: ''}
+				{@const cell_color =
+					column.contentType === 'colorscale' && is_nonzero && column.colorPalette
+						? column.scaleColumn
+							? color_scale(row[column.scaleColumn]).hex()
+							: color_scale(row[column.id]).hex()
+						: ''}
+				{@const font_color = column.redNegatives
+					? row[column.id] < 0
+						? 'rgb(220 38 38)'
+						: ''
+					: column.contentType === 'colorscale' && is_nonzero && column.colorPalette
+						? chroma.contrast(cell_color, uiColours.grey999) <
+							chroma.contrast(cell_color, 'white') + 0.5
+							? 'white'
+							: uiColours.grey999
+						: ''}
+				{@const bottom_border =
+					i !== displayedData.length - 1 &&
+					rowLines &&
+					column.contentType === 'colorscale' &&
+					is_nonzero &&
+					column.colorPalette
+						? `1px solid ${chroma(cell_color).darken(0.5)}`
+						: ''}
```

Comment:
Not addressable in this PR, but this really is starting to smell like it needs to be it's own component, rather than all of these `{@const}`s